src/features/views/hooks/useItems.ts

```diff
@@ -0,0 +1,92 @@
+import useViewTree from './useViewTree';
+import { FutureBase, IFuture, ResolvedFuture } from 'core/caching/futures';
+import { ZetkinView, ZetkinViewFolder } from '../components/types';
+
+export interface ViewBrowserFolderItem {
+  id: number | string;
+  type: 'folder';
+  title: string;
+  owner: string;
+  data: ZetkinViewFolder;
+  folderId: number | null;
+}
+
+export interface ViewBrowserViewItem {
+  id: number | string;
+  type: 'view';
+  title: string;
+  owner: string;
+  data: ZetkinView;
+  folderId: number | null;
+}
+
+type ViewBrowserBackItem = {
+  folderId: number | null;
+  id: string;
+  title: string | null;
+  type: 'back';
+};
+
+export type ViewBrowserItem =
+  | ViewBrowserFolderItem
+  | ViewBrowserViewItem
+  | ViewBrowserBackItem;
+
+export default function useItems(
```

Comment:
I think this name is a little bit too generic. "Item" is a term that can basically mean anything. In this context it refers to an entry in the view browser, i.e. either a view or a folder. But the only hints in the public interface of this hook is the name of the `folderId` parameter (not very obvious) and the fact that it's part of the `features/views` package.

I would rename it to something like `useViewBrowserItems()`.